File: app/src/main/java/com/jimzrt/umsmounter/Tasks/CheckRootTask.java
package com.jimzrt.umsmounter.Tasks;


import com.topjohnwu.superuser.Shell;

public class CheckRootTask extends BaseTask {

    public CheckRootTask(){
        this.name = "Check Root";
        this.description = "Checking root...";
    }
    @Override
    public boolean execute() {
        if(Shell.rootAccess()){
           this.result ="Root working!\n";
            return true;
        } else {
            this.result ="Root not working!\n";
            return false;
        }



    }


}

File: app/src/main/java/com/jimzrt/umsmounter/tasks/CheckRootTask.kt
package com.jimzrt.umsmounter.tasks

import com.topjohnwu.superuser.Shell

class CheckRootTask : BaseTask() {
    override fun execute(): Boolean {
        return if (Shell.rootAccess()) {
            result = "Root working!\n"
            true
        } else {
            result = "Root not working!\n"
            false
        }
    }

    init {
        name = "Check Root"
        description = "Checking root..."
    }
}